opm/simulators/wells/GroupEconomicLimitsChecker.cpp

```diff
@@ -277,16 +306,45 @@ displayDebugMessage(const std::string &msg) const
 
 void
 GroupEconomicLimitsChecker::
-closeWellsRecursive(Group group)
+addPrintMessage(const std::string &msg, const double value, const double limit, const UnitSystem::measure measure)
+{
+    const std::string header = fmt::format(
+        "{}At time = {:.2f} {} (date = {}): Group {} will close because: \n", this->message_separator,
+        this->unit_system_.from_si(UnitSystem::measure::time, this->simulation_time_),
+        this->unit_system_.name(UnitSystem::measure::time),
+        this->date_string_,
+        this->group_.name()
+    );
+    const std::string measure_name(this->unit_system_.name(measure));
+    const std::string message = fmt::format(msg,
+                                         this->unit_system_.from_si(measure, value), measure_name,
+                                         this->unit_system_.from_si(measure, limit), measure_name);
+
+    this->message_ = header;
+    this->message_ += message;
+}
+
+bool
+GroupEconomicLimitsChecker::
+closeWellsRecursive(Group group, int level)
 {
+    bool wells_closed = false;
+
     if (this->debug_) {
         const std::string msg = fmt::format("closing wells recursive : group {} ", group.name());
         displayDebugMessage(msg);
     }
     for (const std::string& group_name : group.groups()) {
         auto next_group = this->schedule_.getGroup(group_name, this->report_step_idx_);
-        closeWellsRecursive(next_group);
+        wells_closed = wells_closed | closeWellsRecursive(next_group, level+1);
+    }
+    std::string indent("  ");
+    for (int i=0; i<level; ++i) indent += std::string("  ");
```

Comment:
You can use the constructor taking a single character and a count to form this string value in one go:
```C++
const auto indent = std::string(2 * level, ' '); // 2 for number of characters in std::string("  ").
```